### PR TITLE
Id 66 auth0 login

### PR DESCRIPTION
**Pull request by wojtek717**

### CODE REVIEW

File: kpz-mpk/kpz-mpk/Map/ViewModels/MapViewModel.swift [hunk 1]
```diff
@@ -8,18 +8,53 @@
...
 import Foundation
 import MapKit
+import Auth0
 
 final class MapViewModel {
   private weak var presenter: MapViewControllerPresenter?
   let locationService = LocationService()
   let incidentApiService: IncidentApiServiceProtocole = IncidentApiService()
-  
   init(presenter: MapViewControllerPresenter?) {
```
Comment (leszko11) [inline kpz-mpk/kpz-mpk/Map/ViewModels/MapViewModel.swift:17]: You should avoid using singletons by calling static value. More appropriate way of using them is by injecting in initializer.
`init(presenter: MapViewControllerPresenter?, sessionManager: SessionManager = SessionManager.shared)`

File: kpz-mpk/kpz-mpk/Map/ViewModels/MapViewModel.swift [hunk 2]
```diff
@@ -8,18 +8,53 @@
...
+    case "segueType":
+      return SessionManager.shared.credentialsManager.hasValid()
+    default:
+      return true
+    }
+  }
+  
+  func auth0Login() {
+    if !SessionManager.shared.credentialsManager.hasValid() {
+      Auth0
```
Comment (leszko11) [inline kpz-mpk/kpz-mpk/Map/ViewModels/MapViewModel.swift:36]: I think it's better to move it to a method in `SessionManager`

File: kpz-mpk/kpz-mpk/Map/ViewControllers/MapViewController.swift [hunk 3]
```diff
@@ -42,10 +43,17 @@ class MapViewController: UIViewController {
     viewModel.centerMapOnUser()
     registerMapAnnotationViews()
     viewModel.displayAnnotations()
+    SessionManager.shared.renewAuth { (error) in
+      if error != nil { print(error!) }
```
Comment (leszko11) [inline kpz-mpk/kpz-mpk/Map/ViewControllers/MapViewController.swift:47]: Is it needed to unwrap error?

File: kpz-mpk/kpz-mpk/Map/ViewControllers/MapViewController.swift [hunk 4]
```diff
@@ -42,10 +43,17 @@ class MapViewController: UIViewController {
     viewModel.centerMapOnUser()
     registerMapAnnotationViews()
     viewModel.displayAnnotations()
+    SessionManager.shared.renewAuth { (error) in
```
Comment (leszko11) [inline kpz-mpk/kpz-mpk/Map/ViewControllers/MapViewController.swift:46]: I think it's better to move it to `SessionManager`

File: kpz-mpk/kpz-mpk/Map/ViewControllers/MapViewController.swift (hunk 3 above)
Comment (wojtek717) [reply kpz-mpk/kpz-mpk/Map/ViewControllers/MapViewController.swift:47]: We get that warning `Expression implicitly coerced from 'Error?' to 'Any'`. I can cast to Any or provide default value.

Comment (leszko11) [reply kpz-mpk/kpz-mpk/Map/ViewControllers/MapViewController.swift:47]: so maybe `if let error = error {}`?

**Pull request merged**